packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts

```diff
@@ -352,13 +352,72 @@ export const a11y: AuditRuleWithSelector[] = [
 	},
 	{
 		code: 'a11y-missing-content',
-		title: 'Missing content on element important for accessibility',
-		message: 'Headings and anchors must have content to be accessible.',
+		title: 'Missing content',
+		message:
+			'Headings and anchors must have an accessible name, which can come from: inner text, aria-label, aria-labelledby, an img with alt property, or an svg with a tag <title></title>.',
 		selector: a11y_required_content.join(','),
 		match(element: HTMLElement) {
 			// innerText is used to ignore hidden text
 			const innerText = element.innerText.trim();
-			if (innerText === '') return true;
+			if (innerText !== '') return false;
+
+			// Check for aria-label
+			const ariaLabel = element.getAttribute('aria-label')?.trim();
+			if (ariaLabel && ariaLabel !== '') return false;
+
+			// Check for valid aria-labelledby
+			const ariaLabelledby = element.getAttribute('aria-labelledby')?.trim();
+			if (ariaLabelledby) {
+				const ids = ariaLabelledby.split(' ');
+				for (const id of ids) {
+					const referencedElement = document.getElementById(id);
+					if (referencedElement && referencedElement.innerText.trim() !== '') return false;
+				}
+			}
+
+			// Check for <img> with valid alt attribute
+			const imgElements = element.querySelectorAll('img');
+			for (const img of imgElements) {
+				const altAttribute = img.getAttribute('alt');
+				if (altAttribute && altAttribute.trim() !== '') return false;
+			}
+
+			// Check for <svg> with valid title
+			const svgElements = element.querySelectorAll('svg');
+			for (const svg of svgElements) {
+				const titleText = svg.querySelector('title');
+				if (titleText && titleText.textContent && titleText.textContent.trim() !== '') return false;
+			}
+
+			const inputElements = element.querySelectorAll('input');
+			for (const input of inputElements) {
```

Comment:
I'm not sure I understand this input part, what kind of HTML is this testing? This rule only checks headings and anchors, so this would test inputs inside those elements?

Reply:
https://github.com/withastro/astro/pull/9567#pullrequestreview-1800115529

@ematipico do we expect image-type input inside a heading or anchor? Sounds like an image-based form button in which case it would be in a form and as a result out of scope for the selector.

Reply:
Yeah it's new to me as well, 

`<a href="/"><input type="image" src="/image.png" alt="accessible name"></a>` the accessible name of the a tag will become test 

similarly for 
`<a href="/"><input type="text" title="axe name"></a>`

also applies to h1s in my testing with VoiceOver and deque/axe-core

Reply:
Okay, that's weird, but if it exists, we must support it